src/async-wrap.h

```diff
@@ -34,34 +34,44 @@ namespace node {
 
 #define NODE_ASYNC_ID_OFFSET 0xA1C
 
-#define NODE_ASYNC_PROVIDER_TYPES(V)                                          \
+#define NODE_ASYNC_NO_CRYPTO_PROVIDER_TYPES(V)                                \
   V(NONE)                                                                     \
-  V(CONNECTION)                                                               \
   V(FSEVENTWRAP)                                                              \
   V(FSREQWRAP)                                                                \
   V(GETADDRINFOREQWRAP)                                                       \
   V(GETNAMEINFOREQWRAP)                                                       \
   V(HTTPPARSER)                                                               \
   V(JSSTREAM)                                                                 \
-  V(PBKDF2REQUEST)                                                            \
   V(PIPECONNECTWRAP)                                                          \
   V(PIPEWRAP)                                                                 \
   V(PROCESSWRAP)                                                              \
   V(QUERYWRAP)                                                                \
-  V(RANDOMBYTESREQUEST)                                                       \
   V(SHUTDOWNWRAP)                                                             \
   V(SIGNALWRAP)                                                               \
   V(STATWATCHER)                                                              \
   V(TCPCONNECTWRAP)                                                           \
   V(TCPWRAP)                                                                  \
   V(TIMERWRAP)                                                                \
-  V(TLSWRAP)                                                                  \
   V(TTYWRAP)                                                                  \
   V(UDPSENDWRAP)                                                              \
   V(UDPWRAP)                                                                  \
   V(WRITEWRAP)                                                                \
   V(ZLIB)
 
+#if HAVE_OPENSSL
+#define NODE_ASYNC_CRYPTO_PROVIDER_TYPES(V)                                   \
+  V(CONNECTION)                                                               \
```

Comment:
Maybe not in thisPR, but we should probably rename this to `SSLCONNECTION` or something like that?